R/configuration-plan.R

```diff
@@ -141,29 +141,34 @@ ConfigurationPlan <- R6::R6Class(
         return(invisible())
       }
       # UTF-8 encoding is assumed for md files in Intro
-      markdownContent <- readLines(introLocation, encoding = "UTF-8")
+      markdownContent <- readLines(introLocation, encoding = "UTF-8", warn = FALSE)
       addTextChunk(fileName = self$getIntroMarkdown(), text = markdownContent, logFolder = logFolder)
       return(invisible())
     },
 
-    #' @description If available, copy files within subfolders of `Content` directory into `<workflowFolder>`
-    copyContentSubFolders = function() {
-      inputContentFolder <- file.path(self$referenceFolder, "Content")
-      # Get only sub folder names and remove "" from the values returned by list.dirs
-      subFolders <- setdiff(list.dirs(inputContentFolder, full.names = FALSE), "")
-      for(subFolder in subFolders){
-        inputContentSubFolder <- file.path(inputContentFolder, subFolder)
-        outputContentSubFolder <- file.path(self$workflowFolder, subFolder)
-        dir.create(outputContentSubFolder, showWarnings = FALSE, recursive = TRUE)
-        # Copy all files from subfolder into report subfolder
-        for(fileName in list.files(inputContentSubFolder)){
-          # With recursive=TRUE, overwrite images
-          file.copy(
-            from = file.path(inputContentSubFolder, fileName),
-            to = file.path(outputContentSubFolder, fileName),
-            recursive = TRUE
+    #' @description If available, copy all files and folders of `Content` directory into `<workflowFolder>`
+    copyContentFiles = function() {
+      # If no Content available, does not need to copy anything
+      if(!dir.exists(file.path(self$referenceFolder, "Content"))){
+        return(invisible())
+      }
+      # Get list of folders within content as well as all the subfolders using recursive option
+      contentSubFolders <- list.dirs(path = file.path(self$referenceFolder, "Content"), full.names = FALSE, recursive = TRUE)
```

Comment:
`file.path(self$referenceFolder, "Content")` is used 4 times in the function.
You could assign it to a varibale first, e.g.
`srcContentFolder = file.path(self$referenceFolder, "Content")`